### DISCUSSION TITLE
What is correct way to rename a circuit?

### DISCUSSION THREAD

**Discussion by ForceConstant** [emporia-vue-local/esphome#250] [General]
In order to get meaningful names in HA, where should rename the circuit? Esphome or in HA, if in HA where?

**Comment by crzykidd** [comment]
I do it in ESPHome code, this way if I remove a device and readd it everything comes back in with the exact same name.   I have done a number of changes to the config file including making all the circuit names variables at the top etc.  Converting to kWh as the default, etc.     You can see a sample here:  https://github.com/crzykidd/esphome-configs/blob/main/vue2-main-panel.yaml   Hope this helps.
